File: papi/lista1/exercicio9.py
while True:
    try:
        altura = float(input("Por favor digite sua altura: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

while True:
    try:
        peso = float(input("Por favor digite seu peso: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

imc = peso / (altura * altura)

print("O IMC é", imc)

print("Sua condição é: ", end="")

if imc < 18.5:
    print("abaixo do peso")
elif imc < 24.9:
    print("peso ideal")
elif imc < 29.9:
    print("levemente acima do peso")
elif imc < 34.9:
    print("obesidade grau I")
elif imc < 39.9:
    print("obesidade grau II (severa)")
else:
    print("obesidade grau III (mórbida)")

File: papi/lista1/exercicio9.kt
package lista1

fun main() {
	println("Vamos calcular um IMC")
    var peso: Double
    var altura: Double

    while (true) {
        try {
            print("Digite a altura em metros: ")
            altura = readLine()!!.toDouble()
            if (altura <= 0) {
                throw NumberFormatException("Altura inválida. Por favor, digite um valor maior que zero.")
            }
            break
        } catch (e: NumberFormatException) {
            println("Erro ao ler o valor. Por favor, digite um número maior que zero, com o ponto como separador decimal.")
        }
    }

    while(true) {
        try {
            print("Digite o peso em kilos: ")
            peso = readLine()!!.toDouble()
            if (peso <= 0) {
                throw NumberFormatException("Peso inválido. Por favor, digite um valor maior que zero")
            }
            break
        } catch (e: NumberFormatException) {
            println("Erro ao ler o valor. Por favor, digite um número maior que zero, com ponto como separador decimal")
        }
    }

    val imc = peso / (altura * altura)
    println("Para os valores passados, o IMC é: $imc")
    print("O resultado da textual de uma avaliação é: ")

    if (imc < 18.5) {
        println("abaixo do peso")          
    } else if (imc < 25) {
        println("peso ideal (parabéns)")
    } else if (imc < 30) {
        println("levemente acima do peso")
    } else if (imc < 35) {
        println("obesidade grau I")
    } else if (imc < 40) {
        println("obesidade grau II (severa)")
    } else {
        println("obesidade grau III (mórbida)")
    }
}